packages/webapp/src/hooks/query/authentication.tsx

```diff
@@ -37,6 +38,64 @@ export const useAuthLogin = (props) => {
   });
 };
 
+
+/**
+ * Authentication OIDC authorize
+ */
+
+export const useAuthOidcAuthorize = (props) => {
+  const apiRequest = useApiRequest();
+
+  return useMutation((values) => apiRequest.post('oidc/authorize', values), {
+    select: (res) => res.data,
+    onSuccess: (data) => {
+      window.location.href = data.data.authorization_url;
+    },
+    ...props,
+  });
+};
+
+/**
+ * Authentication OIDC login
+ */
+
+export const useAuthOidcLogin = (props) => {
+  const apiRequest = useApiRequest();
+
+  return useMutation((values) => apiRequest.post('oidc/login', values), {
+    select: (res) => res.data,
+    onSuccess: (data) => {
+      // Set authentication cookies.
+      setAuthLoginCookies(data.data);
+
+      window.location.href = '/';
+    },
+    ...props,
+  });
+};
+
+/**
+ * Authentication OIDC logout
+ */
+
+export const useAuthOidcLogout = (props) => {
+  const apiRequest = useApiRequest();
+  const { setLogout } = useAuthActions();
+
+  return useMutation((values) => apiRequest.post('oidc/logout', values), {
+    select: (res) => res.data,
+    onSuccess: (data) => {
+      const logoutUrl = data.data.logout_url;
+
+      setLogout(logoutUrl);
+    },
+    onError: (err) => {
+      setLogout();
+    },
```

Comment:
@abouolia Here I have implemented oidc logout such that after successful oidc login, the session end url will be passed to setLogout and redirected in same way else the normal logout will work accordingly. Is the best way to be followed?